aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -264,6 +266,20 @@ public TransferUtility build() {
                     this.s3.setRegion(Region.getRegion(tuConfig.getString("Region")));
                     this.defaultBucket = tuConfig.getString("Bucket");
 
+                    // Checks if awsconfiguration.json has local testing flag to dangerously connect to HTTP endpoint.
+                    // Defaults to false unless specified.
+                    final boolean canConnectToHTTPEndpoint = tuConfig.has(Constants.LOCAL_TESTING_FLAG_NAME) ?
+                            tuConfig.getBoolean(Constants.LOCAL_TESTING_FLAG_NAME) : false;
+
+                    // Mutates AmazonS3Client object to have local endpoint
+                    if (canConnectToHTTPEndpoint) {
+                        this.s3.setEndpoint(Constants.LOCAL_TESTING_ENDPOINT);
+                        this.s3.setS3ClientOptions(S3ClientOptions.builder()
+                                .setPathStyleAccess(true)
```

Comment:
Add a comment for each of these option as to why we are setting these flags to true.

Reply:
will do